### PR TITLE
Remove flake caused failpoint in watch disrupting progress notifies

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @siyuanfoundation @fuweid @MadhavJivrajani

### CODE REVIEW

**Review by ahrtr — APPROVED** [review]

File: tests/robustness/main_test.go [hunk 1]
```diff
@@ -96,8 +96,11 @@ func testRobustness(ctx context.Context, t *testing.T, lg *zap.Logger, s testSce
 		t.Fatal(err)
 	}
 
-	watchProgressNotifyEnabled := c.Cfg.ServerConfig.ExperimentalWatchProgressNotifyInterval != 0
-	validateGotAtLeastOneProgressNotify(t, r.Client, s.watch.requestProgress || watchProgressNotifyEnabled)
+	failpointImpactingWatch := s.failpoint == failpoint.SleepBeforeSendWatchResponse
```
Comment (siyuanfoundation) [inline tests/robustness/main_test.go:99]: this seems to be a important failpoint to test progress notify. Is there plan to investigate further?

Comment (serathius) [reply tests/robustness/main_test.go:99]: No, with progress notifies not being guaranteed in every situation, it's hard to make a strong requirement on them when we use failpoints that directly disrupt their delivery. We can consider revisiting it only after we improve etcd progress guarantees.

**Pull request merged**